File: leetcode/questions/53/java/53.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE, curSum = Integer.MIN_VALUE;
        
        for (int num: nums) {
            curSum = curSum < 0 ? num : num + curSum;
            maxSum = Math.max(curSum, maxSum);
        }

        return maxSum;
    }
}

File: leetcode/questions/53/python/53.py
from typing import List

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        INF = float('inf')
        max_sum = cur_sum = -INF

        for num in nums:
            cur_sum = num if cur_sum < 0 else num + cur_sum
            max_sum = max(max_sum, cur_sum)

        return max_sum